src/app/data-set/data-set.component.ts

```diff
@@ -0,0 +1,86 @@
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+
+import { AddDataSetDialogComponent } from '../add-data-set-dialog/add-data-set-dialog.component';
+
+import {MatDialog, MatDialogConfig} from '@angular/material/dialog';
+
+import {ViewChild} from '@angular/core';
+import {MatPaginator, MatPaginatorIntl} from '@angular/material/paginator';
+import {MatTableDataSource} from '@angular/material/table';
+
+import { RequestService } from '../request/request.service'
+
+interface DataSet {
+  id: number;
+  url: string;
+  instances: any[];
+  state: number;
+}
+
+@Component({
+  selector: 'app-data-set',
+  templateUrl: './data-set.component.html',
+  styleUrls: ['./data-set.component.css']
+})
+
+export class DataSetComponent implements OnInit {
+
+  private dataSets: DataSet[] = [];
+  displayedColumns = ['id', 'url', 'loading'];
+  dataSource = new MatTableDataSource<DataSet>(this.dataSets);
+
+  private paginator: MatPaginator = new MatPaginator(new MatPaginatorIntl(), ChangeDetectorRef.prototype);
+
+  @ViewChild(MatPaginator) set matPaginator(mp: MatPaginator) {
+    this.paginator = mp;
+    this.dataSource.paginator = this.paginator;
+  }
+
+  constructor(private dialog: MatDialog, private requestService: RequestService) {}
+
+  public addDataSets(){
+    let dialogConfig = new MatDialogConfig();
+    dialogConfig.disableClose = true;
+    dialogConfig.autoFocus = true;
+    dialogConfig.height = '450px';
+    dialogConfig.width = '500px';
+    let dialogRef = this.dialog.open(AddDataSetDialogComponent, dialogConfig);
+    dialogRef.afterClosed().subscribe((res: DataSet[]) => this.addResponseDataSets(res));
+  }
+
+  private async addResponseDataSets(sets: DataSet[]) {
```

Comment:
addResponseDataSets is a mysterious name. Try to avoid technical words like request and response in names if we are not working with a service that's dedicated to making requests and handling responses.